internal/bminventory/inventory.go

```diff
@@ -1510,6 +1511,69 @@ func (b *bareMetalInventory) GetClusterDefaultConfig(_ context.Context, _ instal
 	return installer.NewGetClusterDefaultConfigOK().WithPayload(&body)
 }
 
+func getPlatformFromVendor(vendor models.SystemVendor) models.PlatformType {
+	var supportedManufacturers = map[string]models.PlatformType{
+		"VMware, Inc.": models.PlatformTypeVsphere,
+	}
+	return supportedManufacturers[vendor.Manufacturer]
+}
+
+func decideSupportedPlatforms(hostsPlatforms []models.PlatformType, hostsCount int) *[]models.PlatformType {
+	defaultPlatform := models.PlatformTypeBaremetal
+	supportedPlatforms := []models.PlatformType{defaultPlatform}
+
+	// SNO or no hosts
+	if hostsCount == 0 || hostsCount == 1 {
+		return &supportedPlatforms
+	}
+
+	platforms := make(map[models.PlatformType]bool)
+	existsPlatformsSet := make([]models.PlatformType, 0)
+	for _, platform := range hostsPlatforms {
+		if _, exist := platforms[platform]; !exist {
+			platforms[platform] = true
+			if platform != "" {
+				existsPlatformsSet = append(existsPlatformsSet, platform)
+			}
+		}
+	}
+	if len(existsPlatformsSet) != 1 {
+		// different platform types
+		return &supportedPlatforms
+	}
+
+	supportedPlatforms = append(supportedPlatforms, existsPlatformsSet[0])
+	return &supportedPlatforms
+}
+
+func (b *bareMetalInventory) GetClusterSupportedPlatformsInternal(ctx context.Context, params installer.GetClusterSupportedPlatformsParams) (*[]models.PlatformType, error) {
+	cluster, err := b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: params.ClusterID})
+	if err != nil {
+		return nil, err
+	}
+	reply := b.ListHosts(ctx, installer.ListHostsParams{ClusterID: *cluster.ID})
+	hosts := reply.(*installer.ListHostsOK)
```

Comment:
What will happen if the reply is not OK? (internal error)

Reply:
I updated the code to use cluster.Hosts instead